.github/workflows/build-deploy-dev.yml

```diff
@@ -5,42 +5,133 @@ on:
     types: [opened, reopened, synchronize, closed]
     branches:
       - main
+  workflow_dispatch:
+    inputs:
+      bumb:
+        description: "The type of semantic version increment to make. By default it is 'prerelease'."
+        required: false
+        default: "prerelease"
+        type: choice
+        options:
+          - major
+          - premajor
+          - minor
+          - preminor
+          - patch
+          - prepatch
+          - prerelease
+      preId:
+        description: "Identifier to be used to prefix version increments. By default it is 'prerelease' for pull request event and workflow dispatch will follow the last release tag prefix, if any."
+        required: false
+        type: choice
+        options:
+          -
+          - premajor
+          - preminor
+          - prepatch
+          - prerelease
+      releaseType:
+        description: "Select the release type. By default only tag will be created. if you want to create a release with the tag select the type."
+        type: choice
+        required: false
+        options:
+          -
+          - draft
+          - publish
+
 
 jobs:
   # Print variables for logging and debugging purposes
   checkEnv:
     name: Check Env variables
     environment: DEV
     runs-on: ubuntu-latest
-    if: ${{ github.event.pull_request.merged == true}}
+    if: github.event.pull_request.merged == true || github.event_name == 'workflow_dispatch'
```

Comment:
do we need to compare `github.event.pull_request.merged` with `true`?

Reply:
I believe so, that logic was already there, the idea behind the logic is to do the jobs (create a tag, build, deploy...) only on merge, though  the workflow will trigger in all the below scenarios, 
![image](https://user-images.githubusercontent.com/77353155/226002029-f15e73e0-2264-4416-8f73-5ca42b14abad.png)
right @andrewsignori-aot @guru-aot @dheepak-aot

Reply:
Yes @ann-aot that was the idea. The PR can be opened and closed without actually being merged. I am not aware of another way to do some specific action only when the PR is merged.